UpdateService/handler.ts

```diff
@@ -0,0 +1,189 @@
+
+import {
+  checkSourceIpForHandler,
+  clientIPAndCidrTuple as ipTuple
+} from "io-functions-commons/dist/src/utils/source_ip_check";
+
+import { Context } from "@azure/functions";
+import { identity } from "fp-ts/lib/function";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import { ServiceModel } from "io-functions-commons/dist/src/models/service";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredParamMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_param";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { Service } from "../generated/api-admin/Service";
+import { SubscriptionKeys } from "../generated/api-admin/SubscriptionKeys";
+import { ServicePayload } from "../generated/definitions/ServicePayload";
+import { ServiceWithSubscriptionKeys } from "../generated/definitions/ServiceWithSubscriptionKeys";
+import { withApiRequestWrapper } from "../utils/api";
+import { APIClient } from "../utils/clients/admin";
+import { getLogger, ILogger } from "../utils/logging";
+import { ErrorResponses, IResponseErrorUnauthorized } from "../utils/responses";
+import { serviceOwnerCheckTask } from "../utils/subscription";
+
+type ResponseTypes =
+  | IResponseSuccessJson<ServiceWithSubscriptionKeys>
+  | IResponseErrorUnauthorized
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorNotFound
+  | IResponseErrorTooManyRequests
+  | IResponseErrorInternal;
+
+/**
+ * Type of a UpdateService handler.
+ *
+ * UpdateService expects a service_id and a service payload as input
+ * and returns updated service with subscription keys
+ */
+type IUpdateServiceHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  attrs: IAzureUserAttributes,
+  serviceId: NonEmptyString,
+  servicePayload: ServicePayload
+) => Promise<ResponseTypes>;
+
+const logPrefix = "UpdateServiceHandler";
+
+const getSubscriptionKeysTask = (
+  logger: ILogger,
+  apiClient: ReturnType<APIClient>,
+  serviceId: NonEmptyString
+): TaskEither<ErrorResponses, SubscriptionKeys> =>
+  withApiRequestWrapper(
+    logger,
+    () =>
+      apiClient.getSubscriptionKeys({
+        service_id: serviceId
+      }),
+    200
+  );
+
+const getServiceTask = (
+  logger: ILogger,
+  apiClient: ReturnType<APIClient>,
+  serviceId: NonEmptyString
+): TaskEither<ErrorResponses, Service> =>
+  withApiRequestWrapper(
+    logger,
+    () =>
+      apiClient.getService({
+        service_id: serviceId
+      }),
+    200
+  );
+
+const updateServiceTask = (
+  logger: ILogger,
+  apiClient: ReturnType<APIClient>,
+  servicePayload: ServicePayload,
+  serviceId: NonEmptyString,
+  retrievedService: Service
+): TaskEither<ErrorResponses, Service> =>
+  withApiRequestWrapper(
+    logger,
+    () =>
+      apiClient.updateService({
+        body: {
+          ...retrievedService,
+          ...servicePayload,
```

Comment:
can you add a comment with a todo pointing to the relative task on pivotal to override token_name?

Reply:
Added in [5f377a4](https://github.com/pagopa/io-functions-services/pull/68/commits/5f377a49099c30bdca6dbaa2140ae01e448402f4)